modules/gapi/samples/face_detection.cpp

```diff
@@ -0,0 +1,758 @@
+                                       const double,
+                                       const double) {
+              return cv::empty_array_desc();
+    }
+};
+
+G_API_OP(RunNMS,
+         <GFaces(GFaces, double, bool)>,
+         "sample.custom.mtcnn.run_nms") {
+         static cv::GArrayDesc outMeta(const cv::GArrayDesc&,
+                                       const double, const bool) {
+             return cv::empty_array_desc();
+    }
+};
+
+G_API_OP(AccumulatePyramidOutputs,
+         <GFaces(GFaces, GFaces)>,
+         "sample.custom.mtcnn.accumulate_pyramid_outputs") {
+         static cv::GArrayDesc outMeta(const cv::GArrayDesc&,
+                                       const cv::GArrayDesc&) {
+             return cv::empty_array_desc();
+    }
+};
+
+G_API_OP(ApplyRegression,
+         <GFaces(GFaces, bool)>,
+         "sample.custom.mtcnn.apply_regression") {
+         static cv::GArrayDesc outMeta(const cv::GArrayDesc&, const bool) {
+             return cv::empty_array_desc();
+    }
+};
+
+G_API_OP(BBoxesToSquares,
+         <GFaces(GFaces)>,
+         "sample.custom.mtcnn.bboxes_to_squares") {
+         static cv::GArrayDesc outMeta(const cv::GArrayDesc&) {
+              return cv::empty_array_desc();
+    }
+};
+
+G_API_OP(R_O_NetPreProcGetROIs,
+         <GRects(GFaces, GSize)>,
+         "sample.custom.mtcnn.bboxes_r_o_net_preproc_get_rois") {
+         static cv::GArrayDesc outMeta(const cv::GArrayDesc&, const cv::GOpaqueDesc&) {
+              return cv::empty_array_desc();
+    }
+};
+
+
+G_API_OP(RNetPostProc,
+         <GFaces(GFaces, GMats, GMats, double)>,
+         "sample.custom.mtcnn.rnet_postproc") {
+         static cv::GArrayDesc outMeta(const cv::GArrayDesc&,
+                                       const cv::GArrayDesc&,
+                                       const cv::GArrayDesc&,
+                                       const double) {
+             return cv::empty_array_desc();
+    }
+};
+
+G_API_OP(ONetPostProc,
+         <GFaces(GFaces, GMats, GMats, GMats, double)>,
+         "sample.custom.mtcnn.onet_postproc") {
+         static cv::GArrayDesc outMeta(const cv::GArrayDesc&,
+                                       const cv::GArrayDesc&,
+                                       const cv::GArrayDesc&,
+                                       const cv::GArrayDesc&,
+                                       const double) {
+             return cv::empty_array_desc();
+    }
+};
+
+G_API_OP(SwapFaces,
+         <GFaces(GFaces)>,
+         "sample.custom.mtcnn.swap_faces") {
+         static cv::GArrayDesc outMeta(const cv::GArrayDesc&) {
+              return cv::empty_array_desc();
+    }
+};
+
+G_API_OP(Transpose,
+         <cv::GMat(cv::GMat)>,
+         "sample.custom.mtcnn.transpose") {
+          static cv::GMatDesc outMeta(const cv::GMatDesc in) {
+               const cv::GMatDesc out_desc = { in.depth, in.chan, cv::Size(in.size.height,
```

Comment:
This may be a bad example, normally output meta is defined from the input one so you don't copy every field individually like here but alter only what's changing.

```c++
return in.withSize(height, width);
```
is the recommended approach.

Reply:
Done